desktop/ui/src/main/java/org/datacleaner/widgets/tree/SchemaTree.java

```diff
@@ -269,10 +273,22 @@ private DefaultMutableTreeNode createLibrary(final DefaultMutableTreeNode librar
         final Set<ComponentSuperCategory> superCategories = descriptorProvider.getComponentSuperCategories();
         for (ComponentSuperCategory superCategory : superCategories) {
             final DefaultMutableTreeNode schemaNode = new DefaultMutableTreeNode(superCategory);
-            libraryRoot.add(schemaNode);
             final Collection<? extends ComponentDescriptor<?>> componentDescriptors = descriptorProvider
                     .getComponentDescriptorsOfSuperCategory(superCategory);
 
+            final List<ComponentDescriptor<?>> filteredComponentDescriptors = new ArrayList<>();
+
+            for (ComponentDescriptor<?> componentDescriptor : componentDescriptors) {
+                final String displayName = componentDescriptor.getDisplayName();
+                if (displayName.toLowerCase().contains(_searchTerm.toLowerCase())) {
```

Comment:
Could search term ever be null? Maybe it happens for some reason and then your strange issue is that this causes a NPE? Just a thought, not sure at all.